app/controllers/users_controller.rb

```diff
@@ -7,6 +7,10 @@ def edit; end
 
   def update
     batch_number = user_params[:batch_number]&.gsub(/[^\d]/, "")
+    if batch_number.to_i > 2**31 - 1 # max value of a 4-bytes integer column in a database
+      redirect_to settings_path, notice: "Please enter a valid batch number"
+      return
```

Comment:
You can omit this `return` here
The `redirect_to` line above will make a new HTTP request into another controller action, meaning that `return` line would not be reached

Reply:
Careful. `redirect_to` (much like `render`) does not stop the method's execution — it only skips automatic default rendering. The `Batch.find_or_create` and other lines would still be called if the return wasn't here.

Reply:
Uops, my bad! It's been a while